src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentVersionCache.cs

```diff
@@ -1,15 +1,193 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
 using Microsoft.CodeAnalysis.Razor.ProjectSystem;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
-internal abstract class DocumentVersionCache : IProjectSnapshotChangeTrigger
+internal sealed class DocumentVersionCache : IDocumentVersionCache, IProjectSnapshotChangeTrigger
 {
-    public abstract void Initialize(ProjectSnapshotManagerBase projectManager);
+    internal const int MaxDocumentTrackingCount = 20;
 
-    public abstract bool TryGetDocumentVersion(IDocumentSnapshot documentSnapshot, [NotNullWhen(true)] out int? version);
-    public abstract void TrackDocumentVersion(IDocumentSnapshot documentSnapshot, int version);
+    // Internal for testing
```

Comment:
None of this is changed, GitHub just things this is an update, when its actually a rename from DefaultDocumentVersionCache